andisiambuku/Landing-Page-API-Docs
# feat(api): implement Unsplash and Rick and Morty API integrations

**Summary**

This PR introduces the following key features:

Integration with the Unsplash API to fetch photos and its docs.
Implementation of the Rick and Morty API to retrieve character data and its docs.
Comprehensive unit tests for both API integrations using Jest.

**Details**

Unsplash API Integration

Added functions to interact with the Unsplash API:
getAllPhotos: Fetches a list of all photos.
getOnePhoto: Fetches a single photo by ID.
getPhotoStatistics: Fetches statistics for a specific photo.
getRandomPhoto: Fetches a random photo.
Implemented error handling for each function to manage API response failures.
Created comprehensive unit tests for each function to ensure proper functionality and error handling.
Rick and Morty API Integration

Developed the RickMortyData React component to display character data from the Rick and Morty API.
Utilizes the Apollo Client to fetch data with the GET_ALL_CHARACTERS GraphQL query.
Displays a loading state while data is being fetched.
Renders character information including name, status, species, gender, and image.


Added Jest configuration to handle TypeScript and module aliases.
Included test cases to cover:
Successful data fetching and rendering for both APIs.
Handling of loading states.
Error handling scenarios.
Verified the accuracy of API data rendering in the RickMortyData component.


Ensure to configure the environment variables for the Unsplash API access key before running the project.
Follow the setup guide in the README to install dependencies and run tests.

**Test Instructions**

Unsplash API Tests:

Run npm test and verify all Unsplash API tests pass.
Check coverage for getAllPhotos, getOnePhoto, getPhotoStatistics, and getRandomPhoto functions.
Rick and Morty API Component:

Run npm test and verify all RickMortyData component tests pass.
Check that the loading state, character data, and images render correctly.
Manual Verification:

Start the application using npm run dev.
Navigate to the Unsplash and Rick and Morty sections.
Verify data is fetched and displayed correctly.
